src/wasm-linker.cpp

```diff
@@ -297,114 +298,38 @@ void Linker::emscriptenGlue(std::ostream& o) {
     WasmPrinter::printModule(&out.wasm, std::cerr);
   }
 
-  out.wasm.removeImport(EMSCRIPTEN_ASM_CONST); // we create _sig versions
-
-  makeDynCallThunks();
-
-  o << ";; METADATA: { ";
-  // find asmConst calls, and emit their metadata
-  struct AsmConstWalker : public PostWalker<AsmConstWalker, Visitor<AsmConstWalker>> {
-    Linker* parent;
-
-    std::map<std::string, std::set<std::string>> sigsForCode;
-    std::map<std::string, Address> ids;
-    std::set<std::string> allSigs;
+  for (auto f : emscripten::makeDynCallThunks(out.wasm, getTableData())) {
```

Comment:
I think we should maybe just have something like `std::vector<Name>& getTable()` and make its implementation keep the abstraction that pretends there can only ever be one table. If this has the side effect of creating an empty table, then we should just ensure that the other places that care handle that correctly. Also figure out why the segments and offsets, etc. But I think that can be in a future CL and doesn't need to block this one. I think what we have here is still an improvement.